bot/bot.py

```diff
@@ -111,18 +111,27 @@ async def retry_handle(update: Update, context: CallbackContext):
     await message_handle(update, context, message=last_dialog_message["user"], use_new_dialog_timeout=False)
 
 
+async def bot_is_mentioned(update:Update):
```

Comment:
Hi! 

The worst thing about the bot when working with group chats is that even when the privacy setting is enabled, it sees and responds not only to messages addressed to it, but also to messages replying to messages that replied to it, i.e., it sees the whole chain and "sticks its nose in" every time. I modified your function so that the bot only reacts to replies to its own messages since writing a mention is very inconvenient. With this implementation, only reply messages to the bot and mentions will be processed.

```
async def bot_is_mentioned(update: Update, context: CallbackContext):
    message = update.message

    if message.chat.type == "private":
        return True

    if message.entities and message.entities[0].type == 'mention' and message.entities[0].offset == 0:
        return True


    if message.reply_to_message:
        reply = message.reply_to_message
        if reply.from_user.id == context.bot.id and reply.text != message.text:
            return True

    return False
```

I added the 'context' argument to the function to gain access to the bot's ID. You will need to pass 'context' to the function when calling it.